spec/controllers/admin/tag_sets_controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Admin::TagSetsController do
+  let(:current_user) { create :user }
+  let(:tag_set) { create :tag_set }
+
+  it_behaves_like 'it requires login'
+
+  context 'when admin' do
+    before { session[:user] = create :admin }
+
+    describe '#index' do
+      before { get :index }
+
+      it 'is successful' do
+        expect(response).to have_http_status(:success)
+        expect(response).to render_template('index')
+      end
+    end
+
+    describe '#show' do
+      let(:tag_set) { create :tag_set }
+
+      before { get :show, params: { id: tag_set.id } }
+
+      it 'is successful' do
+        expect(response).to have_http_status(:success)
+        expect(response).to render_template('show')
+      end
+    end
+
+    describe '#create' do
+      before { post :create, params: { tag_set: { name: 'test-123', tag_group_id: create(:tag_group).id } } }
+
+      it 'renders the show template' do
```

Comment:
Just wondering whether if it's possible for the `save` function to go wrong (e.g. a validation error, maybe?).

Reply:
Yeah it should show the new template if the params are bad or save fails, ill add a test.